lib/main.dart

```diff
@@ -12,11 +12,26 @@ import 'app/services/navigation.dart';
 import 'app/widgets/hpi_theme.dart';
 
 Future<ByteData> fetchFont(String url) async {
-  final response = await http.get(url);
-  if (response.statusCode == 200)
+  //added handler for network connectivity check
+  try {
+  final result = await InternetAddress.lookup('google.com');
+  if (result.isNotEmpty && result[0].rawAddress.isNotEmpty) {
+    print('connected');
```

Comment:
Whether `google.com` is reachable does not indicate whether `hpi.de` (which is where we download the font) is reachable. However, the easier solution here would be to just try to load the font and fall back on the default if it fails (i.e. exceeds a timeout of 5 seconds) instead of checking connectivity before.

Reply:
why is that downloading form internet on active state of app is necessary? i mean you could download font and move to assets folder (add to `pubspec.yaml`). then you could load directly.

Reply:
Oh, we need to adhere to the corporate identity control committee of the HPI. Because the font is closed-source and copyrighted, we aren't allowed to directly incorporate it into open source projects, including this one. That's why we opted for downloading the font instead.